src/optics/lib/Convex/CardanoApi/Lenses.hs

```diff
@@ -408,16 +408,45 @@ _TxInsCollateralIso = iso from to
     [] -> C.TxInsCollateralNone
     xs -> C.TxInsCollateral C.alonzoBasedEra xs
 
-_TxMintValue :: forall era build. (C.IsMaryBasedEra era) => Iso' (TxMintValue build era) (Map PolicyId [(C.AssetName, Quantity, BuildTxWith build (ScriptWitness WitCtxMint era))])
+_TxMintValue :: forall era build. (C.IsMaryBasedEra era) => Iso' (TxMintValue build era) (Map PolicyId (Map C.AssetName Quantity, BuildTxWith build (ScriptWitness WitCtxMint era)))
 _TxMintValue = iso from to
  where
-  from :: TxMintValue build era -> Map PolicyId [(C.AssetName, Quantity, BuildTxWith build (ScriptWitness WitCtxMint era))]
+  from :: TxMintValue build era -> Map PolicyId (Map C.AssetName Quantity, BuildTxWith build (ScriptWitness WitCtxMint era))
   from = \case
     C.TxMintNone -> mempty
-    C.TxMintValue _ mp -> mp
+    C.TxMintValue _ mp -> fromCapiMap mp
   to mp
     | Map.null mp = C.TxMintNone
-    | otherwise = C.TxMintValue C.maryBasedEra mp
+    | otherwise = C.TxMintValue C.maryBasedEra $ toCapiMap $ removeZeroEntries mp
+
+  toCapiMap =
+    Map.map
+      ( \(innerMap, w) ->
+          map (\(an, q) -> (an, q, w)) $ Map.toList innerMap
+      )
+
+  fromCapiMap =
+    -- Remove 0 quantity entries
+    removeZeroEntries
+      -- Convert to nested map structure
+      . Map.map
+        ( \tups ->
+            foldr
+              ( \(an, q, wit) (m, _) ->
```

Comment:
For clarify, maybe change to `( \(an, q, _) (m, wit) ->` since you're assuming that initial wit you encountered is the correct one? Not mandatory though.

Reply:
I changed the structure and pushed the tuple to the inner map, hopefully that helps with clarity.